heroku/resource_heroku_ssl_test.go

```diff
@@ -62,30 +66,30 @@ resource "heroku_app" "one" {
 }
 
 resource "heroku_slug" "one" {
-    app = "${heroku_app.one.id}"
-    file_path = "test-fixtures/slug.tgz"
-    process_types = {
-      web = "ruby server.rb"
-    }
+  app = heroku_app.one.id
+  file_path = "test-fixtures/slug.tgz"
+  process_types = {
+    web = "ruby server.rb"
+  }
 }
 
 resource "heroku_app_release" "one" {
-  app = "${heroku_app.one.id}"
-  slug_id = "${heroku_slug.one.id}"
+  app = heroku_app.one.id
+  slug_id = heroku_slug.one.id
 }
 
 resource "heroku_formation" "web" {
-  app = "${heroku_app.one.id}"
+  app = heroku_app.one.id
   type = "web"
   size = "hobby"
   quantity = 1
   depends_on = [heroku_app_release.one]
 }
 
 resource "heroku_ssl" "one" {
-  app = "${heroku_app.one.id}"
-  certificate_chain="${file("%s")}"
-  private_key="${file("%s")}"
+  app_id = heroku_app.one.uuid
```

Comment:
I don't think we have a uuid attribute, but the id is a uuid:

```suggestion
  app_id = heroku_app.one.id
```

Reply:
`uuid` is computed attribute of `heroku_app`: https://github.com/heroku/terraform-provider-heroku/blob/master/heroku/resource_heroku_app.go#L176-L178

And `heroku_app.id` is set to the name of the app: https://github.com/heroku/terraform-provider-heroku/blob/master/heroku/resource_heroku_app.go#L321

Oh the joys of two primary keys.

Reply:
oh, wow

Reply:
Yeah, there has been an inconsistency in the provider over what primary key is used as the resource ID. Fixing this consistency would require a some work around adding migrations so users don't end up with mass resource recreations.

For new resources, I've been steering us towards the path of setting `resource.id` to the UUID and then having `resource.name` for convenience.